### ISSUE TITLE
Album artwork grainy for just one album

### ISSUE THREAD

**Issue by n-o-o-b** [vanilla-music/vanilla#752]
- Vanilla Music Version (Settings -> About): 1.0.59
- Android Version: LineageOS 14.1; Android 7.1
- Phone vendor / model: Motorola Moto E 2nd Gen 4G
- Application was downloaded from (use `[x]`):
  - [x] Play Store
  - [ ] F-Droid
  - [ ] Other (Nightly Build)
- Did this work in a previous version?
  - [ ] Yes
  - [ ] No
  - [x] NA / Don't know.

## Description
I've got an album of FLAC files without embedded artwork and a cover.jpg file in the same folder. The file looks crisp on my computer and when I view it in my phone's gallery but when I play the album in Vanilla Music, the artwork looks grainy. The same thing happens for the stock LineageOS music player and Shuttle Music, but not for Poweramp (trial version). The artwork is a 615 KB, 1000x1000 JPG file.
I tried deleting re-copying the whole folder and the artwork file and unticking some of the options in the Cover Art section of the settings and it didn't help. I tried other artwork; it only happens for this file.

## Steps to reproduce the issue
Download the artwork from below and put it into a folder with some music. Open Vanilla Music and play the music.

## Screenshot / Video (If applicable)
The Now Playing screen in Vanilla Music:

![screenshot_20180316-125743](https://user-images.githubusercontent.com/19697175/37522251-47ebe1f8-291b-11e8-817b-0ee1d4651bc4.png)

Viewing the file in my phone's gallery:

![screenshot_20180316-130629](https://user-images.githubusercontent.com/19697175/37522266-52723de8-291b-11e8-92a3-2d69acd1d0c1.png)

The file itself:

![folder](https://user-images.githubusercontent.com/19697175/37522298-68649c72-291b-11e8-9552-3a6ee0446e21.jpg)

**Comment by adrian-bl** [comment]
Works fine for me:

![image](https://user-images.githubusercontent.com/568734/37569794-b4f7b3b2-2ae7-11e8-9ff9-3a540ad346d0.png)

Looks like we got a (poorly) resized version from Androids media library:

I suppose you did try to uncheck the 'Load artwork from Android' option?
Note that you must *kill* Vanilla Music for to see the effect as the poorly resized cover may still be in the in-memory (LRU) cache.

**Comment by n-o-o-b** [comment]
@adrian-bl I did try both of those things. (Killing the app and unticking the option.)
I'm encouraged by the fact that it worked for you but I don't know how to make it work for me. I tried deleting the whole folder, re-copying the art, killing the app between each attempt, re-scanning the media library, reinstalling the app, restarting my device, and I couldn't get it to work.

**Comment by adrian-bl** [comment]
Are you sure that there is no low-res version of the file floating around? (Maybe in a different folder?)

Another thing you could try is:
 * Set 'Group albums by folder' in media scanner settings (to handle files with the same album tag in different folders as distinct albums)
 * Copy the picture and (some) audio files to a new folder
 * Check whether the art is still grainy.

If it is, then something seems to be wrong with the resizing on your device.
If it isn't, it means that we somehow read (and favor) a low-res version

**Comment by n-o-o-b** [comment]
@adrian-bl I don't know whether there's a low-res version anywhere. I didn't put one on my phone, but there could be one created by Android or something (?). I don't really know how it works.

I tried your suggested steps. The album art is still looks the same. However, I came across a new problem, which is that when I take out my SD card to put music on it and put it back into my device, all my music vanished from Vanilla Music and only re-appears if I do a re-scan and tick the option "flush media database".

While I'm here, could you explain what the "Android media library"/"Android media database" is? I can't find an explanation online.

**Comment by adrian-bl** [comment]
> which is that when I take out my SD card to put music on it and put it back into my device, all my music 
> vanished from Vanilla Music and only re-appears if I do a re-scan and tick the option "flush media 
> database".

Yes, that WAI: Android will tell us that the files are gone, so we remove them from the database (as we could not play them)


> While I'm here, could you explain what the "Android media library"/"Android media database" is? I 
> can't find an explanation online.

The Android OS keeps a database with all media files present on the device (photos, ringtones, music). Unlike many other media players, vanilla does *not* use this library but uses its own (more flexible and less buggy) database. We do, however, sometimes read from androids own media library to quickly discover new files.

**Comment by n-o-o-b** [comment]
@adrian-bl I just recently got round to testing this out properly. I think it's an issue with Vanilla. I tried Shuttle Music and Phonograph/Vinyl Music Player and those didn't have the issue. Well, I had to tick their respective options for ignoring the Media Store covers, but after that they worked. I have that option ticked in Vanilla and all my artwork is still low-res. I tried re-scanning (with and without 'flush media database' ticked), grouping albums by folder, killing the app and uninstalling it, and none of it helped.

**Comment by david-sawatzke** [comment]
I've got the same similar issue, with some (but not all) album covers (with covers loading only from folder):

The ones I've found have a resolution higher than my smartphone, so somethings probably going wrong while downscaling. This would also explain why the scaling of the sample file worked on some devices, since it it's resolution isn't that high. 

[cover.tar.gz](https://github.com/vanilla-music/vanilla/files/3648288/cover.tar.gz)

![photo5213095338561351186](https://user-images.githubusercontent.com/11035569/65531304-db5c6800-def9-11e9-8bc0-f874b214a08a.jpg)
![photo5211183403804765143](https://user-images.githubusercontent.com/11035569/65531308-de575880-def9-11e9-868b-d6b2b1aa8d98.jpg)

This is with Vanilla Music 1.0.83 on LineageOS 16.0.
I've also had the same issue with LineageOS 14.0 on a different device.

**Comment by ChiefMikeK** [comment]
Searching for low res? this app worked for me 🐸 :neckbeard: 

- [Unseen Gallery -Cached images & thumbnails Manager - Google Play](https://play.google.com/store/apps/details?id=uni.unseengallery.ke&hl=en_US)
